web/api/common/spotifyclient.go

```diff
@@ -0,0 +1,90 @@
+package common
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type spotifyClient struct {
+	sync.WaitGroup
+	httpClient *http.Client
+}
+
+func newSpotifyClient(httpClient *http.Client) *spotifyClient {
+	return &spotifyClient{
```

Comment:
Client should have timeout property which is `math.Max`'d with `httpClient.Timeout`. If this ends up being `0` (neither are set to anything greater than 0), then there is no timeout (bad idea but should be allowed).

This will likely involve making the `spotifyClient` struct thread-safe (even though the `*http.Client` and `sync.WaitGroup` constituents already are safe).